pkg/controller/container-runtime-config/helpers_test.go

```diff
@@ -502,6 +502,44 @@ func TestUpdatePolicyJSON(t *testing.T) {
 			allowed:       []string{"allow.io"},
 			errorExpected: true,
 		},
+		{
+			name:    "allowed registries: atomic does not allow deep namespaces",
+			allowed: []string{"allow.io", "allowed-example.com/namespace1/namespace2/namespace3"},
+			want: signature.Policy{
+				Default: signature.PolicyRequirements{signature.NewPRReject()},
+				Transports: map[string]signature.PolicyTransportScopes{
+					"atomic": map[string]signature.PolicyRequirements{
+						"allow.io": signature.PolicyRequirements{signature.NewPRInsecureAcceptAnything()},
+					},
+					"docker": map[string]signature.PolicyRequirements{
+						"allow.io": signature.PolicyRequirements{signature.NewPRInsecureAcceptAnything()},
+						"allowed-example.com/namespace1/namespace2/namespace3": signature.PolicyRequirements{signature.NewPRInsecureAcceptAnything()},
+					},
+					"docker-daemon": map[string]signature.PolicyRequirements{
+						"": signature.PolicyRequirements{signature.NewPRInsecureAcceptAnything()},
+					},
+				},
+			},
+		},
+		{
+			name:    "blocked registries: atomic does not allow deep namespaces",
+			allowed: []string{"allow.io", "blocked-example.com/namespace1/namespace2/namespace3"},
```

Comment:
This is not actually testing blocked registries.

(And then `allow.io` is a confusing name for a blocking example.)